website/mailchimp_utils.py

```diff
@@ -76,7 +76,14 @@ def unsubscribe_mailchimp(list_name, user_id, username=None, send_goodbye=True):
     user = OSFUser.load(user_id)
     m = get_mailchimp_api()
     list_id = get_list_id_from_name(list_name=list_name)
-    m.lists.unsubscribe(id=list_id, email={'email': username or user.username}, send_goodbye=send_goodbye)
+    
+    try:
+        m.lists.unsubscribe(
+            id=list_id, email={'email': username or user.username},
+            send_goodbye=send_goodbye
+        )
+    except mailchimp.ListNotSubscribedError:
+        pass
```

Comment:
The error comes form when trying to do a queue task to unsub a user who has already unsubscribed from the mailchimp. Since user has already unsubscribed and all this function to do is unsubscribe user from mailchimp, it does no harm to silently pass the error and allow to update mailing_list user field.